Make it an error (not a deprecation warning) to reference tasks from an included build using finalizedBy, mustRunAfter, or shouldRunAfter (#21434)
gradle/gradle: subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc

. Run `gradle wrapper --gradle-version {gradleVersion}` to update the project to {gradleVersion}.
. Try to run the project and debug any errors using the <<troubleshooting.adoc#troubleshooting, Troubleshooting Guide>>.

[[changes_8.0]]
== Upgrading from 7.6 and earlier

=== Warnings that are now errors

==== Referencing tasks in an included build with `finalizedBy`, `mustRunAfter` or `shouldRunAfter`

> If you use `finalizedBy`, `mustRunAfter` or `shouldRunAfter` to reference tasks contained in an included build, this will now result in an execution time error upon running the referencing tasks.

Reviewer: ```suggestion
Referencing tasks contained in an included build with any of the following methods now results in an execution time error:

- `finalizedBy`
- `mustRunAfter`
- `shouldRunAfter`
```

Suggestion: move the list of methods into a bulleted list so it's easier to read the sentence on its own (and then look at the list of methods to see if you're affected).

Feel free to reword the introduction sentence if my reword isn't totally accurate.

Author: Thanks for the suggestion, I added it [here](https://github.com/gradle/gradle/pull/21511)

Author: @lkasso - some docs feedback here
